stonesoup/models/transition/nonlinear.py

```diff
@@ -1,46 +1,215 @@
+            [sv1[0] + np.sin(dAngle)/turn_rate * sv1[1] - sv1[3] / turn_rate *
+             (1. - np.cos(dAngle)),
+             sv1[1] * np.cos(dAngle) - sv1[3] * np.sin(dAngle),
+             sv1[1] / turn_rate * (1. - np.cos(dAngle)) + sv1[2] + sv1[3] * np.sin(dAngle)
+             / turn_rate,
+             sv1[1] * np.sin(dAngle) + sv1[3] * np.cos(dAngle),
+             turn_rate])
+        if isinstance(noise, bool) or noise is None:
+            if noise:
+                noise = self.rvs(**kwargs)
+            else:
+                noise = 0
+        return sv2 + noise
 
-        Parameters
-        ----------
-        state : :class:`stonesoup.state.State`
-            The state to be transitioned according to the models in :py:attr:`~model_list`.
+    def covar(self, time_interval, **kwargs):
+        """Returns the transition model noise covariance matrix.
 
         Returns
         -------
-        state_vector: :class:`stonesoup.types.array.StateVector`
-            of shape (:py:attr:`~ndim_state, 1`). The resultant state vector of the transition.
+        : :class:`stonesoup.types.state.CovarianceMatrix` of shape\
+        (:py:attr:`~ndim_state`, :py:attr:`~ndim_state`)
+            The process noise covariance.
         """
-        temp_state = copy.copy(state)
-        ndim_count = 0
-        state_vector = np.zeros((self.ndim_state, 1)).view(StateVector)
+        q_x, q_y = self.linear_noise_coeffs
+        q = self.turn_noise_coeff
+        dt = time_interval.total_seconds()
+
+        G2 = np.array([[dt**2/2], [dt]])
+        v = np.array([[q_x], [q_y], [q]])
+        G = block_diag(G2, G2, 1)
+
+        C = G@v@v.T@G.T
+
+        return CovarianceMatrix(C)
+
+
+class ConstantTurnSandwich(ConstantTurn):
+    r"""This is a class implementation of a time-variant 2D Constant Turn
+    Model. This model is used, as opposed to the normal :class:`~.ConstantTurn`
+    model, when the turn occurs in 2 dimensions that are not adjacent in the
+    state vector, eg if the turn occurs in the x-z plane but the state vector
+    is of the form :math:`(x,y,z)`. The list of transition models are to be
+    applied to any state variables that lie in between, eg if for the above
+    example you wanted the y component to move with constant velocity, you
+    would put a :class:`~.ConstantVelocity` model in the list.
+
+    The target is assumed to move with (nearly) constant velocity and also
+    unknown (nearly) constant turn rate.
+    """
+    model_list: Sequence[GaussianTransitionModel] = Property(
+        doc="List of Transition Models.")
+
+    @property
+    def ndim_state(self):
+        """ndim_state getter method
+
+        Returns
+        -------
+        : :class:`int`
+            The number of combined model state dimensions.
+        """
+        return sum(model.ndim_state for model in self.model_list) + 5
+
+    def function(self, state, noise=False, **kwargs) -> StateVector:
+        state_tmp = copy.deepcopy(state)
```

Comment:
Could probably do a regular `copy` here, as long as you replace `state_vector` 3 lines down.